### ISSUE TITLE
✨ Career development form

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#12011]
## ✨ Feature

Inline form for editing a users career development, to be placed on the "Career planning" page.

## 🕵️ Details

This will be a new inline form using the mutations/queries defined in the [backend data for Career planning](https://github.com/GCTC-NTGC/gc-digital-talent/issues/12001) ticket.

The section subheading includes a status icon, showing `error` if any fields are empty, and `success` if all fields are filled.

## 🎨 Design

https://www.figma.com/design/Xnkie0dSq9RujvQRAZwrY4/Employee-profile-(Employees)?node-id=20-3397&t=gS6c1YiJVZrt1vWZ-0
@JoshBeveridge 

## 📸 Screenshot

Please see figma

## 🙋‍♀️ Proposed Implementation

Create a new component under `~/pages/Users/CareerPlanning/components` that uses the `ToggleSection` to include an inline form.

If the page stub is completed, include the component on the page under the second section as the first item.

## ✅ Acceptance Criteria

- [ ] Inline editing form component created
  - [ ] Radio buttons are required, checkboxes or checkbox groups are not
  - [ ] "Missing information" labels appear, the the overall status is incomplete, if any of the fields are _null_. Empty arrays are okay.
  - [ ] In the view mode, checkboxes are represented along with a green/grey icon, radio buttons are not.
- [ ] Connected to API (queries, mutations)
- [ ] If page exists, added to page in correct location
  - [ ] set the status of the Table of Contents entries.
- [ ] Design review by @JoshBeveridge 
- [ ] Language review by @NienkeBr  

## 🛑 Blockers

```[tasklist]
### Blocked By
- [ ] #12001
```

**Comment by tristan-orourke** [comment]
Our current inline form component doesn't exactly match the design in figma. Don't worry about it, use the current component, we will update the component in its own ticket. We will update it everywhere consistently.

**Comment by tristan-orourke** [comment]
We'll make the checkbox groups optional, that seems the easiest way to move forward.
